public/app/features/explore/Wrapper.tsx

```diff
@@ -72,6 +90,10 @@ function Wrapper(props: GrafanaRouteComponentProps<{}, ExploreQueryParams>) {
 
   const hasSplit = Boolean(queryParams.left) && Boolean(queryParams.right);
 
+  if (!correlations) {
```

Comment:
maybe premature optimization, but I'm wondering if we would really want to keep the whole rapper from showing until its loaded all the correlations? would we want this to be per datasource?

Reply:
Had similar thoughts. In theory we could load correlations only after getting results so we don't even need to delay running the queries. Just went with the simplest option for now to see how it evolves, but can be optimised now as well.

Reply:
@Elfo404 / @gelicia would something like this https://github.com/grafana/grafana/pull/56295/commits/ec0bd25bde978debe280635f17135e17ad289a25 make sense 🤔

Reply:
yasss ❤️ 

not sure about the naming of `ensureCorrelations` but i also wouldn't know how to call it :D

Reply:
Thanks for suggesting it 👍  it's basically your idea sprinkled with some stupid function names ❤️